test/functional/inspec_exec_test.rb

```diff
@@ -178,6 +178,25 @@ def stderr
     assert_exit_code 100, out
   end
 
+  it "executes only specified controls when selecting the controls by literal names" do
+    inspec("exec " + File.join(profile_path, "filter_table") + " --no-create-lockfile --controls foo")
+    _(stdout).must_equal "
+Profile: InSpec Profile (filter_table)
+Version: 0.1.0
+Target:  local://
+
+  ✔  foo: a thing
```

Comment:
This is likely too brittle of a test - the check marks vary under Windows, for example. I suggest you write tests that look specifically for what you are looking for. For example, 

```
_(out.stdout).wont_include "bar"
_(out.stdout).wont_include "baz"
```